crates/bifrost/src/watchdog.rs

```diff
@@ -125,11 +204,32 @@ impl Watchdog {
             Some(cmd) = self.inbound.recv() => {
                 self.handle_command(cmd)
             }
+            Some(Ok(failed_trim_requests)) = OptionFuture::from(self.in_flight_trim.as_mut()) => {
+                self.in_flight_trim = None;
+
+                // if trims failed, merge them back into the map for the next attempt
+                for (log_id, trim_request) in failed_trim_requests {
+                    self.store_trim_request(log_id, trim_request)
+                }
```

Comment:
In case of a failed trim, we might want to debounce the next trim attempt a little bit. Otherwise, we might create a busy loop with a trim failing and then on the next loop iteration trying it immediately again.

Reply:
currently the code just drops failed trims, as watchdog operations are supposedly best effort. we could just do that?

Reply:
I think just dropping would be fine – is it something that deserves a metric, otherwise at least a ~info-level log message?